src/ChannelServer/Skills/General/SelfSkillHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Melia.Channel.Network;
+using Melia.Channel.Skills.Base;
+using Melia.Channel.World.Entities;
+using Melia.Shared.World;
+
+namespace Melia.Channel.Skills.General
+{
+	/// <summary>
+	/// Targeted ground skill implementation when the client provides targets
+	/// </summary>
+	public class SelfSkillHandler : ISelfSkillHandler
+	{
+		public void Handle(Skill skill, Character caster, Position castPosition, Direction direction)
+		{
+			if (skill.SpendSp > 0)
```

Comment:
Couldn't you simplify this by just checking whether caster.Sp is lower than SpendSp? Because that will only be the case if SpendSp is greater than 0 and the caster doesn't have enough Sp.

Reply:
That's a great idea, but not for this PR.